### PR TITLE
Add Skylight service for application monitoring

### PR DESCRIPTION
**Pull request by lozette**
## Changes in this PR

Trello: https://trello.com/c/2zO0f36l/573-add-skylight-service-for-application-performance-monitoring

Add the Skylight gem & service to monitor performance.

The `SKYLIGHT_ACCESS_TOKEN` has been added to Travis CI and is in the 1Password
vault for reference.

The `SKYLIGHT_ENV` variable has been set to `staging` for the develop branch on
Travis CI, and `production` for the master branch.

Screenshot 2020-05-01 at 15 06 38

Skylight data will be logged under the `accounts-skylight@dxw.com` account on
https://www.skylight.io/

*Note* I'm not sure all the places I've added `SKYLIGHT_ACCESS_TOKEN` to are still required, I just added it everywhere!

## Next steps

- [x] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [x] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

File: doc/architecture/decisions/0020-use-skylight-application-performance-monitoring.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# 20. use-skylight-application-performance-monitoring
```
Comment (erbridge) [inline doc/architecture/decisions/0020-use-skylight-application-performance-monitoring.md:1]: I'd like some information about what Skylight provides for us in here, and what the other options might have been.

Comment (lozette) [reply doc/architecture/decisions/0020-use-skylight-application-performance-monitoring.md:1]: Added a bit more in!

**Comment by lozette** [comment]
@mec I think that's a good call 👍

**Comment by lozette** [comment]
Screenshot 2020-05-07 at 11 58 27

I've set `SKYLIGHT_ENABLE_SIDEKIQ` to be true for master & develop branches only

**Pull request merged**